lint/rule_panel_rate_interval.go

```diff
@@ -2,50 +2,70 @@ package lint
 
 import (
 	"fmt"
-	"regexp"
+	"time"
+
+	"github.com/prometheus/prometheus/promql/parser"
 )
 
-var rangeVectorRegexp = regexp.MustCompile(`\[([^:)]+)\]`)
+type inspector func(parser.Node, []parser.Node) error
+
+func (f inspector) Visit(node parser.Node, path []parser.Node) (parser.Visitor, error) {
+	if err := f(node, path); err != nil {
+		return nil, err
+	}
+	return f, nil
+}
 
 // NewPanelRateIntervalRule builds a lint rule for panels with Prometheus queries which checks
 // all range vector selectors use $__rate_interval.
 func NewPanelRateIntervalRule() *PanelRuleFunc {
+	rateIntervalMagicDuration, err := time.ParseDuration(globalVariables["__rate_interval"].(string))
+	if err != nil {
+		// Will not happen
+		panic(err)
+	}
 	return &PanelRuleFunc{
 		name:        "panel-rate-interval-rule",
 		description: "Checks that each panel uses $__rate_interval.",
 		fn: func(d Dashboard, p Panel) Result {
 			if t := getTemplateDatasource(d); t == nil || t.Query != "prometheus" {
 				// Missing template datasources is a separate rule.
-				return Result{
-					Severity: Success,
-					Message:  "OK",
-				}
+				return SuccessResult
 			}
 
 			if !panelHasQueries(p) {
 				// Don't lint certain types of panels.
-				return Result{
-					Severity: Success,
-					Message:  "OK",
-				}
+				return SuccessResult
```

Comment:
I like this, but in the future refactors like this should ideally be a separate PR.